feat(replay): add kafka config for new ingest-feedbacks consumer

aliu39 opened the pull request:
Linked to https://github.com/getsentry/sentry/issues/66100 and
https://getsentry.atlassian.net/browse/OPS-5317

Sentry-kafka-schemas (merge second?) [PR](url)
- may have to rename to ingest-feedback-events

May have to wait on:
https://github.com/getsentry/sentry/pull/66288
https://github.com/getsentry/sentry/pull/66381

aliu39 committed:
feat(replay): add kafka config for new ingest-feedbacks consumer

aliu39 committed:
Merge branch 'master' into aliu/ingest-feedbacks-config

lynnagara commented on src/sentry/consumers/__init__.py:
```diff
@@ -269,6 +269,14 @@ def ingest_events_options() -> list[click.Option]:
     },
+    "ingest-feedbacks": {
+        "topic": settings.KAFKA_INGEST_FEEDBACKS,
+        "strategy_factory": "sentry.ingest.consumer.factory.IngestStrategyFactory",
```
Each consumer should have it's own strategy? (the ingest strategy is already doing far more than it should and we should not add additional logic to it)

aliu3ntry replied:
Any tips/references for how to write a new strategy? In ingest/consumer/factory.py, right

lynnagara replied:
The directory structure in sentry is a bit misleading, but the `ingest` module really just refers to the errors/transactions ingest consumer today and not any of the rest of the piplines. I would suggest keeping it either entirely separate, or inside a new module `ingest.user_feedback` or something like that. See https://getsentry.github.io/arroyo/ for docs how to write a strategy.

cmanallen replied:
@lynnagara We're not adding additional logic to it.  This behavior already exists in that consumer.  We're just moving it to a new topic.  I agree we should split out this product into its own consumer.  Can that be handled in a follow-up PR?

aliu3ntry commented on src/sentry/conf/server.py:
```diff
@@ -3452,6 +3452,7 @@ def build_cdc_postgres_init_db_volume(settings: Any) -> dict[str, dict[str, str]
 KAFKA_SESSIONS_SUBSCRIPTIONS_RESULTS = "sessions-subscription-results"
 KAFKA_METRICS_SUBSCRIPTIONS_RESULTS = "metrics-subscription-results"
 KAFKA_INGEST_EVENTS = "ingest-events"
+KAFKA_INGEST_FEEDBACKS = "ingest-feedbacks"
```
This is for events from the [user feedback widget](https://docs.sentry.io/product/user-feedback/), which is owned by the replay team.

lynnagara committed:
Merge branch 'master' into aliu/ingest-feedbacks-config

aliu39 committed:
Merge branch 'master' into aliu/ingest-feedbacks-config

aliu39 committed:
feat(replay): remove organizations:session-replay-event-linking feature flag  (#66257)

removes flag declaration and all refs
https://github.com/getsentry/team-replay/issues/387

k-fish committed:
fix(metrics-extraction): Fix meta lookup in factory (#66256)

Was using TSResult here which is wrong, it's a dict w/ data or a dict of
dicts if it's a group-by. The presence of the data key should be
indicative.

Fixes SENTRY-2V84

k-fish committed:
fix(metrics-extraction): Allow overriding split decision (#66259)

### Summary
If there are any problems with the split decision this will cause it to
always re-run the code and save the new decision,

corps committed:
feat(autofix): timeout with autofix, query state (#66241)

Abdkhan14 committed:
feat(new-trace): Added logic for red errored rows. (#66194)

<img width="1035" alt="Screenshot 2024-03-03 at 9 36 22 PM"
src="https://github.com/getsentry/sentry/assets/60121741/860a6370-b1aa-41dc-b5c8-b62b9778ffa5">

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

Commit:
feat(replay): adds current user to replay in header (#66253)

This PR adds the current user to the header of the replay in the issue
details page:

Before:
![Screenshot 2024-03-04 at 1 15
37 PM](https://github.com/getsentry/sentry/assets/8533851/614047aa-e0da-4520-91d0-5d71f22f8cd9)


After:
![Screenshot 2024-03-04 at 1 15
31 PM](https://github.com/getsentry/sentry/assets/8533851/4592b55d-ac83-4448-8af5-06988dfc5aa1)

wedamija committed:
fix(crons): Remove db constraint from historical migrations for monitor environment (#66265)

These constraints are causing migrations to fail in getsentry, where
we've set up the router to move these tables to a separate database.

azaslavsky committed:
feat(backup): Enable shimming deleted fields (#66248)

We need to support deleted fields on Django model schemas for a number
of self-hosted released (currently 2). This means that even if we delete
a field on HEAD, we still need to make the import script aware that the
field could exit on imported JSON, and pre-process that JSON in such a
way that the field is removed in cases where the JSON was generated by
an old version that still contained the field.

There is no purpose in testing this code standalone. It passes tests in
#65923, which is the first actual use of this shimming mechanism.

lynnagara committed:
ref: Move topics to new-style of Kafka topic definition (#66271)

Just moving 3 consumers as a first step, will migrate the rest as a
follow up

aliu39 committed:
fix(replay): delete flaky test test_get_replays_dead_rage_click_cutoff (#66277)

edwardgou-sentry committed:
fix(performance): Turn off INP on the browser integration sdk (#66279)

Toggles off INP

aliu39 committed:
add ingest-feedbacks to KAFKA_TOPIC_TO_CLUSTER and kafka_definition

evanpurkhiser committed:
ref(crons): Remove unused code branch in mark_failed (#66284)

failure_issue_threshold will never be falsey due because of this
condition here [0]

[0]:
https://github.com/getsentry/sentry/blob/11e039354141a8487776be19d7535d0da2ee53ce/src/sentry/monitors/logic/mark_failed.py#L44-L45

ArthurKnaus committed:
chore(ddm): Remove formula feature flag in BE (#66199)

Commit:
ref(dynamic-sampling): Remove capture message for timing (#66245)

Commit:
feat(ddm): Add support for units normalization in formulas (#66213)

Commit:
fix(ddm): Sample uniformly the list of requested code locations (#66296)

Swatinem committed:
Clean up `ArtifactBundle(FlatFile)Index`, step 2 (#66207)

This is the second step following #66206:

Step 2 of removing outdated tables and columns from the
`ArtifactBundleIndex` involves removing them from the migration state
*only*.

Actual removal from SQL follows in a third step.

ArthurKnaus committed:
ref(ddm): Remove normalization logic (#66132)

Remove normalization logic from FE.

- closes https://github.com/getsentry/sentry/issues/65765

Commit:
feat(ddm): Allow multiple use case ids to be queried at once and parallelize (#66298)

anonrig committed:
perf: use pipeline for redis rate limiter (#64428)

anonrig committed:
perf: do not include projects and teams (#66232)

ArthurKnaus committed:
fix(ddm): Duplicate chart sample ids (#66301)

* Use spanId for identifying chart samples
* Wrap widget with error boundary
* Fix query response meta type

Swatinem committed:
Remove `FileBlob` locking entirely (#66297)

Instead of using a Redis-based lock to protect concurrent `FileBlob`
uploads and deletes, the code was already changed previously to handle
the cases of concurrent uploads, as well as to reduce the likelihood of
concurrent deletes by regularly bumping the `FileBlob` timestamp.

The lock has been disabled via an `option` recently, and now it is time
to completely remove this code.

gggritso committed:
ref(perf): Introduce `MetricReadout` component (#66254)

Extracted from #66229 as a first step. The `MetricReadout` component
represents a single metric block in our various Starfish metrics
ribbons.

This component has some advantages over using plain `Block`s:

- consistent loading state
- consistent missing data state
- built-in support for different kinds of units
- built-in simple support for tooltips

This makes it easy to use it in all kinds of contexts.

antonpirker committed:
Bump Sentry Python SDK to version 2.0.0a2 (#66052)

These are changes necessary to run `sentry` with the Python SDK 2.0.0a2.

Three tests are still failing and I commented them out to be able to
test this on S4S and Canary.

The package `pytest-sentry` is not yet compatible to SDK 2.0, we will
release an updated version of it soon. After this has been released we
can upgrade `sentry` to SDK 2.0 for good.

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>
Co-authored-by: Francesco Vigliaturo <francesco.vigliaturo@sentry.io>
Co-authored-by: ArthurKnaus <arthur.knaus@sentry.io>

k-fish committed:
fix(metrics-extraction): Fix attribute vs dict lookup (#66306)

### Summary
Missed that the inner block was also getting `data` as if it was an
attribute.

Fixes SENTRY-2VPR

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

getsentry-bot committed:
Revert "Bump Sentry Python SDK to version 2.0.0a2 (#66052)"

This reverts commit b9b433b2cc955f42820c1ad26a3476d7fbb93989.

Co-authored-by: antonpirker <202325+antonpirker@users.noreply.github.com>

Commit:
fix(ddm): Fix units normalization when * and / are used (#66304)

edwardgou-sentry committed:
fix(performance): Fix a typo in the mri of measurements.inp (#66273)

`measurements.inp` should point to `d:spans/webvital.inp@millisecond`

Commit:
fix(ddm): Handle division by zero (#66311)

getsentry-bot committed:
Revert "ref(sdk): Bump js sdk version to 7.105.0 (#66224)"

This reverts commit e968207b755a35211808a5535b4aae887929c385.

Co-authored-by: edwardgou-sentry <83961295+edwardgou-sentry@users.noreply.github.com>

Zylphrex committed:
fix(profiling): Add missing transaction condition for profile digest (#66309)

This was querying all profiles accidentally without the transaction
condition.

loewenheim committed:
feat(proguard): Add Symbolicator.process_jvm method (#66064)

This is part of the effort to port proguard processing to Symbolicator
(https://github.com/getsentry/team-processing/issues/116). In
particular, it hooks into the new `symbolicate-jvm` endpoint introduced
in https://github.com/getsentry/symbolicator/pull/1393.

markstory committed:
fix(hybridcloud) Add visiblity to region configuration (#66295)

When we add regions we need a way to keep the region out of UI
interactions and the public API responses until it has been validated.

The `find_all_multitenant_region_names` method is used in sentry and
getsentry to get the list of multi-tenant regions that are available for
new account creation.

Once we have a new region functionally working it can be made visible.
Even after a region is visible account creation would be gated by
feature flags.

dashed committed:
fix(hybrid-cloud): Prevent API gateway from resolving redirects (#66291)

DominikB2014 committed:
fix padding (#66318)

Just a quick padding fix on the resource page, nothing fancy.

Before

![image](https://github.com/getsentry/sentry/assets/44422760/ff0097a0-5309-43e1-bea3-744daf827e28)


After

![image](https://github.com/getsentry/sentry/assets/44422760/4e0adbfc-87f6-48cd-bb6a-9acae370a3dc)

edwardgou-sentry committed:
feat(sdk): bumps js sdk to 7.103.0 and updates browsertracingintegration (#66317)

bumps js sdk to 7.103.0 and updates browsertracingintegration

Zylphrex committed:
feat(indexed-spans): Support profile.id on indexed spans (#66178)

This column is called `profile.id` on other datasets. Not sure why it's
called `profile_id` here. Especially when all other id columns in this
dataset uses the dot notation. Keeping `profile_id` around for
compatibility.

---------

Co-authored-by: William Mak <william@wmak.io>

gggritso committed:
fix(perf): Fix HTTP widget documentation link (#66011)

Fixes #65071

Directs users with no HTTP span to the "Set Up Performance" page rather
than the "Integrations" page which doesn't even exist for a lot of
platforms.

markstory committed:
feat(processing) Move redis operations in reprocessing to servicewrapper (#66168)

We need to shift the reprocessing workload to a new redis cluster
without downtime. In order to do that I would like to use our Service
wrappers and ServiceDelegator to dual write and then shift reads via
runtime options.

Before I can move load with ServiceDelegator the existing logic needs to
be moved into a service wrapper.

evanh committed:
ref(metrics): Change aggregation options to be a dictionary (#65666)

Change the get_aggregation_options to return a dictionary of options,
and
associate the options with a time window.

This is precursor work to being able to check the aggregation options
when
querying, to ensure queries are not asking using options that are not
available.

evanpurkhiser committed:
ref(crons): Remove legacy fingerprint generation (#66250)

While implementing monitor incidents there was a period of time where we
were "cutting over" to monitors creating incident entities instead of
just producing issue occurrences where the grouphash was computed from
an interim 'last_state_change'. All monitors are now producing incidents
when they fail, so we no longer need the fallback grouphash computation

The `last_state_change` was only required for this interim period and
will also be removed in a future commit.

ceorourke committed:
fix(daily summary): Fix subquery, exit early if no users (#66261)

Fixes SENTRY-2VF1 and SENTRY-2VF2

malwilley committed:
feat(issue-priority): Rename issue stream tab inbox -> priority and remove unresolved tab (#66269)

lynnagara committed:
ref: Move more code to the new way of defining kafka topics and overrides (#66283)

This moves the `dlq_topic` consumer definitions and anything that uses
the `get_topic_definition` helper function over to the new style kafka
topic config.

scttcper committed:
feat(issues): Remove quicktrace component, move tracelink (#66333)

edwardgou-sentry committed:
feat(performance): Updates useIndexedSpans to accept an optional fields array (#66275)

Updates useIndexedSpans to accept an optional fields array to allow
selecting specific fields.
Refactors existing call to use new function signature.

michellewzhang committed:
ref(feedback): toggles are false by default (#66331)

https://github.com/getsentry/sentry/assets/56095982/e3ee988f-893a-4f5d-b8e7-827e12291a40

malwilley committed:
feat(issue-priority): Use 'Med' label in priority pills (#66272)

Also adjusts the stream column width since we don't need as much space
any longer

lynnagara committed:
ref: Move query subscriptions over to new-style kafka config (#66287)

Depends on https://github.com/getsentry/sentry/pull/66283

Zylphrex committed:
styles(metrics): Use disabled button for no profile (#66341)

The no value message is quite long so use a disabled button for it to
appear more compact.

snigdhas committed:
ref(issues): Add typing to issue status and issue platform files (#66093)

michellewzhang committed:
ref(feedback): update links and system default for onboarding (#66330)

Changed snippet example to `colorScheme: "system"`:
<img width="457" alt="SCR-20240305-jdjj"
src="https://github.com/getsentry/sentry/assets/56095982/fe99f6aa-8b3e-4ed6-b117-15cacf9273dd">

Updated links and hyperlink to the specific "bring your own button"
section in the docs:
<img width="465" alt="SCR-20240305-jcve"
src="https://github.com/getsentry/sentry/assets/56095982/2a9984a4-4947-437c-bcc5-b93d991a70b9">

leeandher committed:
chore(commit-context): Remove unused feature flag (#66345)

With https://github.com/getsentry/sentry/pull/65346 and
https://github.com/getsentry/getsentry/pull/12983 merged, self-hosted
and all organizations on SaaS now have access to the commit-context
feature for suspect commits. The feature flag can be safely removed.

evanpurkhiser committed:
ref(crons): Remove usage of last_state_change (#66328)

This code is no longer needed now that we have stopped generating legacy
event hashes in GH-66250

schew2381 committed:
fix(staff): Set access by calling parent method before checking staff (#66266)

john-z-yang committed:
fix(metrics): Update indexer telemetry to handle no messages (#66350)

### Overview

Update indexer telemetry to handle no messages in case everything is
killswitched

ryan953 committed:
feat(discover): Remove replay_id column from default discover table view (#66346)

Fixes https://github.com/getsentry/sentry/issues/66322

malwilley committed:
fix(ui): Remove custom prism text selection styles (#66240)

The existing styles were too similar to the highlighted line color.
Using browser default coloring works well enough.

evanpurkhiser committed:
ref(crons): Use `id__in` query for failed check-in occurrence creation (#66327)

Follow up of
https://github.com/getsentry/sentry/pull/66250/files#r1512012737

GabeVillalobos committed:
fix(hc): Provide region specific urls for chunk-upload config (#66320)

michellewzhang committed:
ref(feedback): add rust to platforms (#66356)

<img width="459" alt="SCR-20240305-lrbx"
src="https://github.com/getsentry/sentry/assets/56095982/90706ed5-cf5d-4996-a9e1-79b94f38e7d2">

getsentry-bot committed:
Revert "ref: Move more code to the new way of defining kafka topics and overrides (#66283)"

This reverts commit f7ffe5a3d31a87ac334c191cc8eb9550d2e1ebce.

Co-authored-by: ayirr7 <47572810+ayirr7@users.noreply.github.com>

getsentry-bot committed:
Revert "ref: Move query subscriptions over to new-style kafka config (#66287)"

This reverts commit f9c2ffd3926a9215f245e34e0dae2f0f7134a3fb.

Co-authored-by: IanWoodard <17186604+IanWoodard@users.noreply.github.com>

evanpurkhiser committed:
ref(crons): Drop last_state_change column (#66353)

This is no longer needed after usage removal in GH-66328

nhsiehgit committed:
bump more_slow_alerts to 300 alerts (#66367)

[per
discussions](https://sentry.slack.com/archives/C06J6HKFKLG/p1708464708399359)

bumping the 'more slow alerts' group to 300 alerts.

This is a temporary fix and will need to be addressed properly
(potentially via
[inc-666](https://getsentry.atlassian.net/browse/INC-666), but there may
be other solutions here as well)

nhsiehgit committed:
Activated alert rule apis feature flag (#66361)

Includes feature flag and monitor type validator in the alert rule api
serializer

Serializer converts int value into enum

ceorourke committed:
ref(daily summary): Add logger when sending summary (#66369)

Add a log line when we send the notification as I'm doing internal
testing.

malwilley committed:
feat(issue-priority): Add last edited by info and feedback button to dropdown (#66362)

evanpurkhiser committed:
ref(crons): Clean up mark_failed logic (#66372)

Just some minor re-organiztion to make it a bit more readible

ryan953 committed:
docs: Wrap stories with OrganizationContainer (#66323)

<img width="687" alt="SCR-20240305-ihne"
src="https://github.com/getsentry/sentry/assets/187460/e15745d8-ea24-4ce6-a2ec-858a6c50d1b8">

Commit:
logging(group-owner): adds logging for create group owners (#66371)

Used for debugging race conditions of when we create the row vs when we
check for it

wedamija committed:
fix(crons): Fix backfill script to handle the case where a monitor with a slug already exists in the target org (#66340)

There are 4 monitors that exist twice in the same project, but in
different orgs. The ones that have a different org than the project's
org are totally broken and can't receive checkins. We'll just delete
them, they're inaccessible at the moment.

Commit:
logging(codeowners): adds logging when we delete code group owners (#66358)

Debug logging

michellewzhang committed:
ref(feedback): remove duplicate native platform (#66384)

accidentally had two `native` platforms in this list

malwilley committed:
feat(issue-details): Improved UI for viewing object/array values (#66153)

Fixes https://github.com/getsentry/sentry/issues/58573,
https://github.com/getsentry/sentry/issues/48480

Previously, objects/arrays were only collapsible after they reached
maxDefaultDepth. The major change here is that now each level is
collapsible

- Each level is now always collapsible. Before, objects/arrays were only
collapsible after a certain depth
- Arrays/objects begin collapsed when there are more than 5 items, in
addition to when they are at a certain depth
- Changed the appearance and location of the toggle buttons to better
match commonly-used json viewers
- Added `n items` preview text for collapsed objects/arrays (which is
clickable!)

lynnagara committed:
ref: Move code to new-style kafka topics - take 2 (#66381)

Brings back https://github.com/getsentry/sentry/pull/66283 with fixes to
the indexer strategy. It crashed on the prior deploy as we were not
producing to the correctly resolved topic.

scttcper committed:
test(ui): Add mock date test helpers, remove fake timers (#66342)

wedamija committed:
chore(crons): Return a better message from deprecated crons ingest apis (#66388)

<!-- Describe your PR here. -->

ArthurKnaus committed:
ref(ddm): Remove multiple metrics meta queries (#66305)

Remove request duplication logic for metrics meta queries to utilize
parallelization on the BE and unblock other requests in the browser.

ArthurKnaus committed:
fix(loader-script-setting): Cache update (#66396)

antonpirker committed:
Bump Sentry Python SDK to version 2.0.0a2 (#66397)

New PR for another load test to check performance of the new Python SDK
2.0

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

getsentry-bot committed:
Revert "Bump Sentry Python SDK to version 2.0.0a2 (#66397)"

This reverts commit 51ae02e46645b11c7d2463d8ef0bc6c60e542981.

Co-authored-by: antonpirker <202325+antonpirker@users.noreply.github.com>

ArthurKnaus committed:
feat(ddm): Filter empty series from summary table (#66400)

Workaround as the endpoint returns empty series for groups that are
excluded by a filter.

Swatinem committed:
Clean up `ArtifactBundle(FlatFile)Index`, step 3 (#66208)

This is the third step, following
https://github.com/getsentry/sentry/pull/66206 and
https://github.com/getsentry/sentry/pull/66207:

In step 3, we fully drop the obsolete columns and tables after they have
been removed from the migration state in prior steps.

ArthurKnaus committed:
feat(ddm): Hiding queries (#66398)

Allow toggling queries from being displayed by clicking their query
symbol.


- closes https://github.com/getsentry/sentry/issues/66201

ArthurKnaus committed:
fix(ddm): Show samples only on selected widget (#66404)

iker-barriocanal committed:
ref(deps): Bump sentry-relay to v0.8.48 (#66401)

Along the way of bumping dependencies, some tests are fixed to cover for
the changes in Relay.

narsaynorath committed:
feat(discover): Remove commented out replay ID column (#66409)

markstory committed:
chore(hybridcloud) Remove unused parameter from sent_incident_alert_notification (#66113)

Complete work started in #65736, #65791, and #65883 to remove a
parameter from the send_incident_alert_notification RPC method.

Fixes HC-1123
Fixes SENTRY-2QAG

markstory committed:
fix(hybridcloud) Make bitbucket uninstall silo safe (#66354)

Follow up on a TODO that has triggered an error. Add tests so that
future maintenance on the bitbucket integration is a bit easier.

Fixes SENTRY-2TNS

markstory committed:
fix(hybridcloud) Add logging for bad RPC requests (#66357)

Add logging when we have a bad request from an RPC method. We don't
often get sentry errors from these failures, and not being able to see
the response of these operations makes debugging harder.

billyvg committed:
feat(replay): Add feature flag for mobile replay player (#66276)

This adds a feature flag to allow support of mobile replay player.
Currently for internal testing only

shruthilayaj committed:
feat(spans): Tag perf problems created from standalone spans (#66325)

As we set up the application logic to run performance issue detection
on standalone spans, we want to collect metrics to check if transaction
based and span based perf issue detection are equivalent. Tagging
metrics with `is_standalone_spans` to track this.
This also updates fingerprint so the standalone span occurrences 
don't get grouped with existing perf issues during testing.

Swatinem committed:
Remove rollout options for Rust Enhancers (#65966)

We have fully enabled these, so lets remove all the options usage
related to it.

This is pretty much a followup from #65533.

Commit:
feat(profiling): pass internal (profile-scoped) dsn for function metrics ingestion to vroom (#66300)

evanh committed:
fix(mql): Handle strings and formula with scalars bug (#66324)

This PR fixes two bugs. Firstly, if a user passed an MQL string into the
layer
(which is completely valid) the layer would throw an exception.

Secondly, if a formula had a component that was itself a formula of
scalars,
e.g. sum(x) + (7 * 24), an exception would be thrown.

markstory committed:
feat(hybridcloud) Add region tag to errors/transactions (#66370)

Add sentry_region to the tags of errors/transactions we collect. This
helps with diagnosing which region an error is occurring in. Previously
we were planning on using `environment` for this but it would have
required a non-trivial amount of monitoring changes. The `sentry_region`
name mirrors the tag used in other monitoring tools we use.

gggritso committed:
feat(perf): Add HTTP module domain overview page scaffold (#66359)

Just some very very basic content to get some content and routing up.

k-fish committed:
revert(sdk): Move us back to 7.102 js sdk (#66412)

…integration (#66317)"

This reverts commit f4c49786a4da31143cf50fab44c41f7e37f47dbc.

Looking for the reason for the drop in pageloads.

michellewzhang committed:
ref(onboarding): refactor useCurrentProjectState to be generalized (#66118)

- DDM, Replays, and Feedback onboarding were all using some variation of
this hook: `useCurrentProjectState`. They're all pretty similar, so I
refactored them into one general reusable hook
- I modified the respective `sidebar.tsx` files to use this new hook
- I also added a hook test

getsentry-bot committed:
Revert "ref(onboarding): refactor useCurrentProjectState to be generalized (#66118)"

This reverts commit 03677ee022b291e55c6ec5c7f46af63379ae9375.

Co-authored-by: michellewzhang <56095982+michellewzhang@users.noreply.github.com>

MichaelSun48 committed:
feat(code-mappings): LA automatic code mapping support for PHP projects (#66151)

(This feature is in LA) 

Whitelisted php projects for automatic code mapping support and added
unit tests

---------

Co-authored-by: Bartek Ogryczak <bartek.ogryczak@sentry.io>
Co-authored-by: Armen Zambrano G. <44410+armenzg@users.noreply.github.com>

schew2381 committed:
chore(superuser): Add logs to check u2f (#66393)

evanpurkhiser committed:
ref(crons): Avoid extra hashing on fingerprint UUIDs (#66373)

michellewzhang committed:
ref(onboarding): refactor useCurrentProjectState to be generalized (#66425)

- had to revert the original PR
(https://github.com/getsentry/sentry/pull/66118) due to a failing test -
fixed in this PR! this is the only change from the previous PR
- DDM, Replays, and Feedback onboarding were all using some variation of
this hook: `useCurrentProjectState`. They're all pretty similar, so I
refactored them into one general reusable hook
- I modified the respective `sidebar.tsx` files to use this new hook
- I also added a hook test

Zylphrex committed:
fix(metrics): Adjust bounding boxes based on operation (#66339)

The bounding boxes right now always bound the metric on the min/max. But
depending on the operation, we should look at different values to bound
determine if it lies in the bounding box. For example, if graphing the
min, we should ensure the min is with in the bounding box.

schew2381 committed:
chore(superuser): Add loggers to _needs_validation (#66364)

There's a bug where we're no longer requiring superuser access category
and reason validation, which is caused by
https://github.com/getsentry/sentry/pull/66043/files.

I want to validate what those booleans are in order to find out if this
is the cause of the bug.

edwardgou-sentry committed:
feat(discover): allows filtering metric queries on span.op (#66423)

Adds `span.op` to `DEFAULT_METRIC_TAGS` so that it is available in
discover metrics queries.

michellewzhang committed:
ref(replay): see full replay in rage click issue goes to breadcrumbs tab (#66374)

From a rage click issue, "see full replay" takes you to the breadcrumbs
tab with the rage & dead click filter selected.

Relates to https://github.com/getsentry/team-replay/issues/394. Temp
solution until we can load in rage click issues to the errors tab


https://github.com/getsentry/sentry/assets/56095982/cda9511d-8458-4fc4-a506-0b36ed217d51

gggritso committed:
ref(perf): Remove usage of generic-ish `SpanMetricsRibbon` (#66415)

Another refactor peeled away from #66229. The `SpanMetricsRibbon`
component is used in two places:

1. The top right of the Query Summary page
2. The completely unused generic span summary page

The ribbon is a little silly because it supports just these two use
cases! This is much better done using the new `MetricReadout` by using
it directly in the parent. No visual changes.

lynnagara committed:
ref: Remove old style topic definition from ingest consumer code (#66352)

settings.KAFKA_INGEST_EVENTS, settings.KAFKA_INGEST_ATTACHMENTS and
settings.KAFKA_INGEST_TRANSACTIONS are deprecated and will be removed.

michellewzhang committed:
feat(feedback/issues): add sidebar to issue details UF empty state (#66351)

- add a sidebar popup when "set up now" is clicked
- add crash report modal onboarding config for javascript with widget
callout

<img width="1072" alt="SCR-20240305-ocgr"
src="https://github.com/getsentry/sentry/assets/56095982/a9b92ffb-7ea0-4bd7-83b0-299e3ae2a2fe">

JoshFerge committed:
fix(feedback): correct usage of option name (#66430)

- I messed up the name of the project option in the endpoint, it should
be plural, not singular
- I was using the wrong project option flag in the check 🤦‍♂️

Thank you to @jas-kas for catching this bug. 🙏🏼 this is what EA is for!


I do want to refactor project options to make it easier to test // less
prone to bugs when adding additional ones. will add a TODO item.

JoshFerge committed:
ref(replays): remove rage click creation celery task (#66360)

We've rolled out rage click_creation_with_replay_event, so we can remove
this celery task and the usages of it. huzzah!

https://sentry.sentry.io/dashboard/77863/?project=1&statsPeriod=24h
(can see bottom metric has dropped to zero, while one to left has gone
up).

this represents that now the new function is being called 100% of the
time so we can remove this code.

wmak committed:
fix(queries): Add more errors to queryhandler (#66432)

- Add urllib3 errors to the query handler
- Mostly downstream hiccups in either case
- Fixes SENTRY-2W3A
- Fixes SENTRY-2TE9
- Fixes SENTRY-2VVR
- Fixes SENTRY-2JBW

wmak committed:
fix(profiling): Add handle_query_errors to profiling (#66433)

- This was missing a handle_query_errors so exceptions were bubbling up
too high
- Fixes SENTRY-2KVM

Commit:
feat(crons): Add new broken detection model (#66375)

Spun off from: https://github.com/getsentry/sentry/pull/65963

scttcper committed:
test(ui): Switch children to react nodes (#66390)

- adds a missing children prop, swaps ReactElement to ReactNode
- removes a render function used as a child

scttcper committed:
ref(issues): Add object type to KeyValue data item (#66391)

scttcper committed:
ref(ui): Remove render function from dropdown label (#66389)

Commit:
feat(crons): Show project information in details area for crons (#65944)

As we transition to a state where monitor slugs are unique to the
project and not the org, we need to have project slug information in the
crons listing page to differentiate duplicate cron slugs within an org.

<img width="1246" alt="image"
src="https://github.com/getsentry/sentry/assets/9372512/f31e4f75-cdc0-41d3-820e-a2928562fe73">

If the project + schedule text gets too long, then it wraps to the next
line. We have a good amount of space in the details area so opted for
wrapping vs. ellipsizing

evanpurkhiser committed:
ref(js): useOrganization in ProjectBadge (#66442)

evanpurkhiser committed:
ref(crons): Drop last_state_change from db (#66429)

No longer needed after GH-66353

edwardgou-sentry committed:
feat(webvitals): Updates web vital module query hooks to actually query for INP measurements (#66421)

Updates all metrics based queries in the Web Vitals module to properly
query for INP measurements and aggregates.
We previously subbed in FID metrics for INP for testing/development
purposes. This PR removes that since INP is now properly queryable in
Discover.

edwardgou-sentry committed:
feat(webvitals): update calculatePerformanceScoreFromStoredTableDataRow to properly account for weights from different score profiles (#66434)

Update calculatePerformanceScoreFromStoredTableDataRow to properly
account for weights from different score profiles

Zylphrex committed:
feat(metrics): Support paginating samples via random seed (#66440)

To give the illusion of a consistent paging, we take the hash of the
limit and offset as the seed to `groupArraySample`.

edwardgou-sentry committed:
feat(webvitals): Greys out FID in the performance score breakdown chart and updates tooltip to label as deprecated (#66438)

Greys out FID in the performance score breakdown chart and updates
tooltip to label as deprecated

edwardgou-sentry committed:
feat(webvitals): updates webvitals landing page slideout panel to support displaying inp (#66439)

updates webvitals landing page slideout panel to support displaying inp

k-fish committed:
ref(sdk): Bump js sdk to 7.105 again (#66436)

### Summary
We're bumping again since we've determined that pageloads are now lower
due to inheriting the sampling decision from the sentry project, along
with the release, public key and transaction name potentially changing

lynnagara committed:
ref: Move the rest of the consumer definitions to the new style (#66343)

JoshFerge committed:
fix(event_manager): catch kafka exceptions in perf issue production (#66447)

catch KafkaExceptions here, as we shouldn't crash the task. Can follow
up with diagnosing the KAFKA_TOO_LARGE payloads and making an upstream
fix.

Should also determine if we should catch this logic in the occurrence
production code instead, so other upstream places don't crash because of
this.

FIxes https://sentry.sentry.io/issues/5040716121/

nhsiehgit committed:
ref: split incident and alert_rule models into separate files (#66386)

currently the incidents/models.py file is nearly 1k lines long and the
file name `model` does not do much to tell us what it's contents are

PR to split up the file names and directory

edit:

oh lawd. i swear most of these are modifying imports now that the
directory structure has been modified

Resolves https://github.com/getsentry/getsentry/pull/13170
Requires https://github.com/getsentry/getsentry/pull/13170

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

wmak committed:
fix(ondemand): Wrong column usage here was breaking (#66437)

- If fields had `-` in them the query would fail since we weren't
resolving the column to a tag correctly
- Fixes: SENTRY-2W5S

wmak committed:
fix(event-stats): Remove error from top events (#66455)

- Fixes SENTRY-2VV0
- Fixes a bug where we were saying errors supports top events even
though it doesn't

iamrajjoshi committed:
feat(api): Allowlist for API Pagination Check (#66463)

* Just the allowlist part of the API Pagination Check
* Need to break down the larger PR because adding `requires` in the PR
description doesn't seem to be working

evanpurkhiser committed:
ref(crons): Move grouphash generation to a field default (#66456)

Let's centralize this here, no need for callers to need to think about
this detail

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

evanpurkhiser committed:
ref(crons): Clarify why incident creation skip happens (#66462)

Guard statments tend to be clearer

gggritso committed:
feat(perf): Record occurrence payload size if it's over the limit (#66460)

Follow-up to #66447

Moves the handling down per comment, adds the size of the payload.

roggenkemper committed:
chore(inbound-filters): Remove feature flag for default inbound filters (#65989)

this pr removes the feature flag for default inbound filters for new
javascript projects now that it is GA'ed

lynnagara committed:
ref: Remove references to deprecated settings (#66355)

settings.KAFKA_OUTCOMES and settings.KAFKA_OUTCOMES_BILLING are
deprecated and will be removed soon

lynnagara commented on src/sentry/consumers/__init__.py:
```diff
@@ -275,6 +275,14 @@ def ingest_events_options() -> list[click.Option]:
+        "strategy_factory": "sentry.ingest.consumer.factory.IngestStrategyFactory",
+        "click_options": ingest_events_options(),
+        "static_args": {
+            "consumer_type": "events",
```
I think you might want to make this a new type. We don't really want to run all the `events` logic, and i'm guessing this will diverge in the future?

scttcper committed:
test(ui): Vendor `react-select-event`, use RTL user events (#66441)

aliu39 committed:
rename to ingest-feedback-events

aliu3ntry commented:
my bad 😂. reopened at https://github.com/getsentry/sentry/pull/66484
